src/tests.ts

```diff
@@ -78,30 +71,49 @@ async function deployContracts(arweave, walletA, walletB) {
     settings: [["isTradeable", true]],
   };
 
-  const contractA = await createContract(
-    arweave,
-    walletA.jwk,
-    contractSrc,
-    JSON.stringify(initialStateA)
-  );
+  let contractATxId, contractBTxId;
 
-  const contractB = await createContract(
-    arweave,
-    walletA.jwk,
-    contractSrc,
-    JSON.stringify(initialStateB)
-  );
+  ({contractTxId: contractATxId} = await warp.createContract.deploy({
+    wallet: walletA.jwk,
+    initState: JSON.stringify(initialStateA),
+    src: contractSrc
+  }));
 
-  await mine();
+  ({contractTxId: contractBTxId} = await warp.createContract.deploy({
+    wallet: walletB.jwk,
+    initState: JSON.stringify(initialStateB),
+    src: contractSrc
+  }));
+
+  await warp.testing.mineBlock();
+
+  const contractA = warp.contract(contractATxId).setEvaluationOptions({
+    internalWrites: true
+  });
+
+  const contractB = warp.contract(contractBTxId).setEvaluationOptions({
+    internalWrites: true
+  });
 
   return {
     contractA,
     contractB,
   };
 }
 
-async function createPair(arweave, walletA, contractA, contractB) {
-  const pairTx = await arweave.createTransaction(
+async function createPair(walletA: JWKInterface, contractA: Contract, contractB: Contract) {
+
+  // creates a pair on "ContractB" from "walletA"
+  // note: "writeInteraction" in "local" env. makes automatic block mining
+  const {originalTxId} = await contractB
+    .connect(walletA)
```

Comment:
`connect(walletA)` effectively means that the transaction with contract will be signed by wallet passed to `connect` function.